lib/plugins/sfDoctrinePlugin/lib/task/sfDoctrineBuildTask.class.php

```diff
@@ -130,7 +130,9 @@ protected function execute($arguments = [], $options = [])
             throw new InvalidArgumentException(sprintf("You must include one or more of the following build options:\n--%s\n\nSee this task's help page for more information:\n\n  php symfony help doctrine:build", join(', --', array_keys($this->getBuildOptions()))));
         }
 
-        if (self::BUILD_DB == (self::BUILD_DB & $mode)) {
```

Comment:
From here down, things get a little more complicated.

On line 129, the options provided to the task are parsed through `$this->calculateMode($options)`, which does bitwise comparison against known class constant values to work out which of the subset of tasks the options require the task to complete.

What that means for us is the **really** dangerous tasks are still lurking in here, and it feels really worrisome they're so close to being executed in production. With that in mind, I've added the `isProduction()` test here too, and blocked out the tasks which absolutely cannot ever be run in production, just in case.